tests/testthat/test-RunAnalyses.R

```diff
@@ -0,0 +1,40 @@
+library(testthat)
+
+test_that("multiple analyses", {
+  # Analysis.R is checked elsewhere
+  exposureOutcome1 <- createExposureOutcome(767410, 444382)
+  exposureOutcome2 <- createExposureOutcome(1314924, 444382)
+  exposureOutcome3 <- createExposureOutcome(907879, 444382)
+  exposureOutcomeList <- list(exposureOutcome1, exposureOutcome2, exposureOutcome3)
+
+  runSelfControlledCohortArgs1 <- createRunSelfControlledCohortArgs(firstExposureOnly = FALSE)
+  runSelfControlledCohortArgs2 <- createRunSelfControlledCohortArgs(firstExposureOnly = TRUE)
+  sccAnalysis1 <- createSccAnalysis(analysisId = 1,
+                                    runSelfControlledCohortArgs = runSelfControlledCohortArgs1)
+  sccAnalysis2 <- createSccAnalysis(analysisId = 2,
+                                    runSelfControlledCohortArgs = runSelfControlledCohortArgs2)
+  sccAnalysisList <- list(sccAnalysis1, sccAnalysis2)
+
+  outputFolder <- file.path(tempdir(), getOption("dbms"))
+  dir.create(outputFolder)
+    withr::defer({
+    unlink(outputFolder, force = TRUE)
+  }, testthat::teardown_env())
+
+  rr <- runSccAnalyses(connectionDetails = connectionDetails,
+                       cdmDatabaseSchema = cdmDatabaseSchema,
+                       oracleTempSchema = oracleTempSchema,
+                       sccAnalysisList = sccAnalysisList,
+                       exposureOutcomeList = exposureOutcomeList,
+                       outputFolder = outputFolder,
+                       computeThreads = 8)
+
+  expect_is(rr, "data.frame")
+  expect_true(file.exists(file.path(outputFolder, "resultsReference.rds")))
+  apply(rr, 1, function(item) {
+    expect_true(file.exists(file.path(outputFolder, item["sccResultsFile"])))
+  })
+
+  result <- summarizeAnalyses(rr, outputFolder)
```

Comment:
Should you also check that the actual results in some way? Something like `expect_true(all(result$irr > 0.1 & result$irr < 5))`

Reply:
Many results will be Inf as there will by outputs with 0 exposures and 0 outcomes. I will try and think of a way to test the actual process with hand generated in a future version.